banhmi/src/services/userService.ts

```diff
@@ -26,14 +26,47 @@ export const getUserFromSocketId = async (socketId) => {
   }
 };
 
-export const updateSocketId = async (userId, socketId) => {
+export const partialUpdateUser = async (req) => {
+  const { socketId, spaceId } = req.body;
+  const userId = req.user.id;
```

Comment:
would be better if using middleware to validate the json schema. if non-undefined fields  are in wrong types, they could cause errors too